src/System.Memory/src/System/ReadOnlySpan.cs

```diff
@@ -177,7 +177,7 @@ internal ReadOnlySpan(Pinnable<T> pinnable, IntPtr byteOffset, int length)
         /// <exception cref="System.IndexOutOfRangeException">
         /// Thrown when index less than 0 or index greater than or equal to Length
         /// </exception>
-        public T this[int index]
+        public ref T this[int index]
```

Comment:
@jaredpar @VSadov this would be `readonly ref` in the future right? This one probably shouldn't be `ref T`?